components/org.wso2.carbon.identity.conditional.auth.functions.user/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/user/IsMemberOfAnyOfGroupsFunctionImpl.java

```diff
@@ -66,6 +77,57 @@ public boolean isMemberOfAnyOfGroups(JsAuthenticatedUser user, List<String> grou
         return result;
     }
 
+    /**
+     * Check the federated user belongs to one of the groups specified in the list of user groups.
+     *
+     * @param user Authenticate user.
+     * @param groupNames      List of groups.
+     * @return whether the federated user belongs to one of the groups specified in the list of user groups .
+     */
+    private boolean isFederatedUserMemberOfAnyGroup(JsAuthenticatedUser user, List<String> groupNames) {
+
+        String[] groupsOfFederatedUser = null;
+        String groupsClaimValue = null;
+        String federatedIdPName = user.getWrapped().getFederatedIdPName();
+        String tenantDomain = user.getWrapped().getTenantDomain();
+        try {
+            // Get the claim URI for groups claim from the claim mappings for federated idp.
+            String groupsClaimURI = Utils.getGroupsClaimURIByClaimMappings(federatedIdPName, tenantDomain);
+            if (groupsClaimURI == null) {
+                groupsClaimURI = GROUPS_LOCAL_CLAIM_URI;
```

Comment:
Can we assume this if the IDP is not of type OIDC?

Reply:
Resolved by https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/132/commits/f35c8722ba2d594ed1c8d5e2cf30c05a3d2f4244